community/security/src/test/java/org/neo4j/commandline/admin/security/SetPasswordCommandTest.java

```diff
@@ -19,97 +19,132 @@
-import static org.junit.Assert.assertThat;
-import static org.junit.Assert.fail;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.neo4j.test.assertion.Assert.assertException;
 
 public class SetPasswordCommandTest extends UsersCommandTestBase
 {
+
+    private UsersCommand usersCommand;
+    private Config config;
+    private File file;
+
+    @Before
+    @Override
+    public void setup()
+    {
+        super.setup();
+        usersCommand = new UsersCommand( homeDir.toPath(), confDir.toPath(), out );
+        config = usersCommand.loadNeo4jConfig();
+        file = BasicAuthManagerFactory.getInitialUserRepositoryFile( config );
+    }
+
     @Test
     public void shouldFailSetPasswordWithNoArguments() throws Exception
     {
-        UsersCommand usersCommand = new UsersCommand( homeDir.toPath(), confDir.toPath(),out );
-
         String[] arguments = {"set-password"};
-        try
-        {
-            usersCommand.execute( arguments );
-            fail( "Should have thrown an exception." );
-        }
-        catch ( IncorrectUsage e )
-        {
-            assertThat( e.getMessage(), containsString( "username" ) );
-        }
+        assertException( () -> usersCommand.execute( arguments ), IncorrectUsage.class, "username" );
     }
 
     @Test
     public void shouldFailSetPasswordWithOnlyOneArgument() throws Exception
     {
-        UsersCommand usersCommand = new UsersCommand( homeDir.toPath(), confDir.toPath(),out );
-
         String[] arguments = {"set-password", "neo4j"};
-        try
-        {
-            usersCommand.execute( arguments );
-            fail( "Should have thrown an exception." );
-        }
-        catch ( IncorrectUsage e )
-        {
-            assertThat( e.getMessage(), containsString( "password" ) );
-        }
+        assertException( () -> usersCommand.execute( arguments ), IncorrectUsage.class, "password" );
     }
 
     @Test
-    public void shouldFailSetPasswordWithNonExistingUser() throws Exception
+    public void shouldAllowSetPasswordWithPasswordChangeRequiredFlag() throws Exception
     {
-        UsersCommand usersCommand = new UsersCommand( homeDir.toPath(), confDir.toPath(), out );
-
-        String[] arguments = {"set-password", "nosuchuser", "whatever"};
-        try
-        {
-            usersCommand.execute( arguments );
-            fail( "Should have thrown an exception." );
-        }
-        catch ( CommandFailed e )
-        {
-            assertThat( e.getMessage(), containsString( "does not exist" ) );
-        }
+        String[] arguments = {"set-password", "neo4j", "123", "--requires-password-change"};
+        usersCommand.execute( arguments );
     }
 
     @Test
-    public void shouldRunSetPasswordWithDefaultUser() throws Throwable
+    public void shouldCreateInitialUserWithPasswordChangeRequired() throws Throwable
```

Comment:
The set-password command cannot create new users. It can only set passwords. The fact that the default user 'neo4j' is _created_ when setting up the database is an implementation detail. We could just as easily ship pre-configured auth and roles files with default users and roles. Treat 'neo4j' as already existing from the users perspective.